lib/test_result.dart

```diff
@@ -75,6 +76,88 @@ class TestResult {
     return TestAttachment.fromJson(response);
   }
 
+  static Future<TestResults> getTestResults({
+    required int testId,
+    int? limit,
+    int? offset,
+  }) async {
```

Comment:
This reads strangely. We know we're getting test results as it's TestResult. But we don't know where we're filtering from. I think renaming the method to more closely reflect its intent would be good, intuitive documentation.
```suggestion
  static Future<TestResults> getFromTest(int testId, {
    int? limit,
    int? offset,
  }) async {
```

Reply:
The logic inside TestRails TestResults is "interesting", but will try to find a bit more related name.